src/proto/api_messages_test.js

```diff
@@ -15,90 +15,110 @@
  */
 
 import {
-    AnalyticsContext,
-    AnalyticsRequest,
-    AnalyticsEvent,
-    deserialize} from './api_messages';
+  AnalyticsContext,
+  AnalyticsEventMeta,
+  AnalyticsRequest,
+  AnalyticsEvent,
+  EventOriginator,
+  deserialize} from './api_messages';
 
 /**
- * Compare two protos
- * @param {!Array} thisArray
- * @param {!Array} otherArray
- * @return {boolean}
- */
+* Compare two protos
+* @param {!Array} thisArray
+* @param {!Array} otherArray
+* @return {boolean}
+*/
 function isEqual(thisArray, otherArray) {
-  if (!otherArray || !thisArray) {
-    return false;
-  }
-  for (let i = 0; i < otherArray.length; i++) {
-    if (Array.isArray(thisArray[i])) {
-      if (!Array.isArray(otherArray[i])) {
-        return false;
-      }
-      const arr = thisArray[i];
-      const otherArr = otherArray[i];
-      if (arr.length != otherArr.length) {
-        return false;
-      }
-      for (let j = 0; j < arr.length; j++) {
-        if (arr[j] != otherArr[j]) {
-          return false;
-        }
-      }
-    } else {
-      if (thisArray[i] != otherArray[i]) {
+if (!otherArray || !thisArray) {
```

Comment:
indented unnecessarily

Reply:
I seemed to see not enough indenting.  I've fixed that